templates/networkpolicies.yaml

```diff
@@ -51,6 +52,9 @@ spec:
           port: 8080
         - protocol: TCP
           port: 8443
+  # Explicitly allow all egress traffic
+  egress:
```

Comment:
I think this rule would override anything admins set, since Kubernetes applies the union of all policies. Adding this means that admins have no way of locking down egress from coderd without managing network policies themselves. Is that the behavior we want?

>Network policies do not conflict; they are additive. If any policy or policies apply to a given pod for a given direction, the connections allowed in that direction from that pod is the union of what the applicable policies allow. Thus, order of evaluation does not affect the policy result.

https://kubernetes.io/docs/concepts/services-networking/network-policies/

My understanding was that the default behavior is to allow all egress, especially if we don't specify policyTypes: Egress - the same page says:

>By default, a pod is non-isolated for egress; all outbound connections are allowed. A pod is isolated for egress if there is any NetworkPolicy that both selects the pod and has "Egress" in its policyTypes; we say that such a policy applies to the pod for egress. When a pod is isolated for egress, the only allowed connections from the pod are those allowed by the egress list of some NetworkPolicy that applies to the pod for egress. The effects of those egress lists combine additively.

Reply:
> admins have no way of locking down egress from coderd without managing network policies themselves. Is that the behavior we want?

I think that's reasonable behaviour. If admins want to override them, they can disable our network policies and manage network policies themselves.